docs/specs/access-control.md

```diff
@@ -0,0 +1,55 @@
+# Access Control
+
+## Information
+
+Information is the information of the user.
+
+Currently there are
+
+- Standard Attributes
+- Custom Attributes
+- Claims
+
+## Party
+
+Party is someone that have access to Information.
+
+Currently there are
+
+- OAuth Client
+- The User on web UI
+- The Caller of Admin API
+
+## Overview of Access Control between Information and Party
+
+|                       |Standard Attributes|Custom Attributes|Claims                               |
+|-----------------------|-------------------|-----------------|-------------------------------------|
+|OAuth Client           |No access          |No access        |Read access controlled by OAuth scope|
+|The User on web UI     |Read and Write     |No access (yet)  |No access                            |
```

Comment:
Would there be a situation like if user A logged in with a OAuth Client ID, which has no access to a "claimA" which is based on "attrA".

But the user on the Web UI have access to "attrA"?

Or on the other hand, would it be technically possible to identify if a user on web UI was logged in from which OAuth Client ID? And is it useful to give them different access based on which Client ID they logged in?

Or maybe too complicated...

Reply:
> Would there be a situation like if user A logged in with a OAuth Client ID, which has no access to a "claimA" which is based on "attrA".
> 
> But the user on the Web UI have access to "attrA"?

In OAuth model, the user is authorizing access to their resources on the resource server to a third party (i.e. the Client), so the user should have access to their resources on the resource server. In our case, that means the user have access to "attrA" on the web UI.

However, if we allow custom attributes to be accessible by the user, the situation becomes more complicated. The user is no longer the sole owner of the resources on the resource server. The developer can define some custom attributes that cannot be read by the user, but at the same time, the developer can map them as claims.
During the OAuth authorization flow, only the user is involved in deciding which scope is granted to the Client.

For example, the developer defines a custom attribute "rbac" that is not readable by the user. At the same time, the developer maps custom attribute "rbac" to claim "app:rbac", with the restriction that scope "app:rbac" is required.
The user cannot read it but they are responsible for granting access to it during authorization flow, which is quite contradictory.

> Or on the other hand, would it be technically possible to identify if a user on web UI was logged in from which OAuth Client ID? And is it useful to give them different access based on which Client ID they logged in?

Actually the relationship is reverse. A IdP session may be associated with many refresh tokens. Strictly speaking, the web UI should not be aware of which client ID. It is because it is perfectly fine to login directly, without any Client. In that case, we show all information of the user. Technically speaking, on the web UI we are only seeing the IdP session cookie, which solely means the user is authenticated, not specific to any Client.

Imagine there are 2 apps using Google Login, you won't expect to see different information on myaccount.google.com after you login one of the app.

Reply:
> The user cannot read it but they are responsible for granting access to it during authorization flow, which is quite contradictory.

Hence I asked the question. I think in that case we might better define the permission as:

1. Claims via each Client ID (under OAuth)
2. Attributes via Setting Page
3. Attributes via GraphQL Endpoint (assume it is admin so all R/W access)

So in future, we might have more permissions such as, if we have user roles: 
* Attributes via User Roles on Management Portal
* Attributes via User Roles on Setting Page